be/src/connector/hive_connector.cpp

```diff
@@ -95,43 +96,78 @@ Status HiveDataSource::open(RuntimeState* state) {
-    if (state->query_options().__isset.enable_scan_datacache) {
-        _use_datacache &= state->query_options().enable_scan_datacache;
-    }
-    if (hdfs_scan_node.__isset.datacache_options &&
-        hdfs_scan_node.datacache_options.__isset.enable_populate_datacache) {
-        _enable_populate_datacache = hdfs_scan_node.datacache_options.enable_populate_datacache;
-    } else if (state->query_options().__isset.enable_populate_datacache) {
-        _enable_populate_datacache = state->query_options().enable_populate_datacache;
-    }
-    if (state->query_options().__isset.enable_datacache_async_populate_mode) {
-        _enable_datacache_aync_populate_mode = state->query_options().enable_datacache_async_populate_mode;
-    }
-    if (state->query_options().__isset.enable_datacache_io_adaptor) {
-        _enable_datacache_io_adaptor = state->query_options().enable_datacache_io_adaptor;
-    }
-    if (state->query_options().__isset.datacache_evict_probability) {
-        _datacache_evict_probability = state->query_options().datacache_evict_probability;
-    }
-    if (state->query_options().__isset.datacache_priority) {
-        _datacache_priority = state->query_options().datacache_priority;
-    }
-    if (state->query_options().__isset.datacache_ttl_seconds) {
-        _datacache_ttl_seconds = state->query_options().datacache_ttl_seconds;
-    }
-    if (state->query_options().__isset.enable_dynamic_prune_scan_range) {
-        _enable_dynamic_prune_scan_range = state->query_options().enable_dynamic_prune_scan_range;
+    // Check that the system meets the requirements for enabling DataCache
+    if (config::datacache_enable && BlockCache::instance()->available()) {
+        // setup priority & ttl seconds
+        int8_t datacache_priority = 0;
+        int64_t datacache_ttl_seconds = 0;
+        if (state->query_options().__isset.datacache_priority) {
+            datacache_priority = state->query_options().datacache_priority;
+        }
+        if (state->query_options().__isset.datacache_ttl_seconds) {
+            datacache_ttl_seconds = state->query_options().datacache_ttl_seconds;
+        }
+
+        if (state->query_options().__isset.enable_cache_select && state->query_options().enable_cache_select) {
+            // set datacache options for cache select
+            _datacache_options = DataCacheOptions{.enable_datacache = true,
+                                                  .enable_cache_select = true,
+                                                  .enable_populate_datacache = true,
+                                                  .enable_datacache_async_populate_mode = false,
+                                                  .enable_datacache_io_adaptor = false,
+                                                  .modification_time = _scan_range.modification_time,
+                                                  .datacache_evict_probability = 0,
```

Comment:
The evict probability should be 100. If set it as 0, the write will failed if the cache is full because it can not evict any old cache items.

Reply:
done